packages/argon2/src/lib.rs

```diff
@@ -181,7 +183,13 @@ impl Task for RawHashTask {
   type JsValue = Buffer;
 
   fn compute(&mut self) -> Result<Self::Output> {
-    let salt = SaltString::generate(&mut OsRng);
+    let generated_salt = SaltString::generate(&mut OsRng);
```

Comment:
The default salt generation can be skipped in case a custom salt is provided

Reply:
the problem is that the generator needs to be out of the if scope otherwise the memory is not persisted, and I unfortunately don't have enough knowledge in rust to do it properly, but i agree with your comment